File: frontend/lib/models/conta.dart
class Conta {
  final int id;
  final String nome;
  final String tipo; // 'banco' | 'especie'
  final double saldoInicial;
  final bool ativo;
  final DateTime criadoEm;
  final DateTime atualizadoEm;
  final double saldoAtual; // derivado, vem do backend

  Conta({
    required this.id,
    required this.nome,
    required this.tipo,
    required this.saldoInicial,
    required this.ativo,
    required this.criadoEm,
    required this.atualizadoEm,
    required this.saldoAtual,
  });

  factory Conta.fromJson(Map<String, dynamic> j) => Conta(
    id: j['id'] as int,
    nome: j['nome'] as String,
    tipo: j['tipo'] as String,
    saldoInicial: (j['saldoInicial'] as num).toDouble(),
    ativo: j['ativo'] as bool,
    criadoEm: DateTime.parse(j['criadoEm'] as String),
    atualizadoEm: DateTime.parse(j['atualizadoEm'] as String),
    saldoAtual: (j['saldoAtual'] as num).toDouble(),
  );

  bool get isEspecie => tipo == 'especie';
  bool get isBanco => tipo == 'banco';
}

File: backend/FluxoCaixa.Api/Models/Conta.cs
namespace FluxoCaixa.Api.Models;

public class Conta
{
    public int Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public TipoConta Tipo { get; set; }
    public decimal SaldoInicial { get; set; }
    public bool Ativo { get; set; } = true;
    public DateTime CriadoEm { get; set; }
    public DateTime AtualizadoEm { get; set; }
}
